gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py

```diff
@@ -0,0 +1,821 @@
+    )
+
+    logger.info("Add all other variant annotations from release HT (dropping freq)")
+    mt = mt.annotate_rows(**variant_annotation_ht[mt.row_key])
+    mt = mt.drop("was_split", "a_index")
+
+    # TODO: lowqual variants how to handle them on both sparse and dense, dense is easy because I can
+    #  remove the variants easily, sparse is more difficult, maybe just make sure annotation is on the variant HT,
+    #  which at the moment is not the case because we just remove them.
+    mt = hl.experimental.densify(mt)
+    mt = mt.filter_rows((~info_ht[mt.row_key].AS_lowqual) & (hl.len(mt.alleles) > 1))
+
+    return mt
+
+
+def main(args):
+    hl.init(log="/subset.log", default_reference="GRCh38")
+
+    if args.create_sample_meta:
+        meta_ht = prepare_sample_annotations()
+        meta_ht.write(release_subset_annotations(subset="hgdp_1kg").path)
+
+    if args.export_meta_txt:
+        release_subset_annotations(subset="hgdp_1kg").ht().export(
+            release_subset_sample_tsv(subset="hgdp_1kg")
+        )
+
+    if args.create_subset_sparse_mt:
+        # NOTE: no longer filtering to high_quality by request from Alicia Martin, but we do filter to variants in
+        # high_quality samples, so how to handle that in the future?
+        meta_ht = release_subset_annotations(subset="hgdp_1kg").ht()
+        mt = get_gnomad_v3_mt(
+            key_by_locus_and_alleles=True, remove_hard_filtered_samples=False
+        )
+
+        logger.info("Filtering MT columns to HGDP + TGP samples")
+        mt = mt.filter_cols(
+            (meta_ht[mt.col_key].subsets.hgdp | meta_ht[mt.col_key].subsets.tgp)
+        )
+        mt = mt.annotate_rows(
+            _telomere_or_centromere=hl.is_defined(
+                telomeres_and_centromeres.ht()[mt.locus]
+            ),
+        )
+        # Filter to rows with a non ref GT outside of telomeres and centomeres, or if there is END info
+        mt = mt.filter_rows(
+            (~mt._telomere_or_centromere & hl.agg.any(mt.LGT.is_non_ref()))
+            | hl.agg.any(hl.is_defined(mt.END))
+        )
+
+        # On rows with END info that fall in telomeres and centomeres, keep only END info, set other entries to missing
+        mt = mt.filter_entries(
+            mt._telomere_or_centromere & hl.is_missing(mt.END), keep=False
+        )
+
+        # Adjust alleles and LA to include only alleles present in the subset
+        mt = adjust_subset_alleles(mt)
+
+        logger.info(
+            "Note: for finalized the HGDP + TGP subset frequency, dense MT, VCFs we adjust the sex genotypes and add "
+            "a fix for older GATK gVCFs with a known depletion of homozygous alternate alleles, and removed standard "
+            "GATK LowQual variants"
+        )
+
+        mt = mt.drop("_telomere_or_centromere")
+        mt.write(
+            release_subset(subset="hgdp_1kg", dense=False).path,
+            overwrite=args.overwrite,
+        )
+
+    if args.create_subset_dense_mt:
```

Comment:
I think a single notification is enough and in case that function gets used outside of this script for some reason it is safer to have there than in main. That function is really only going to ever be used in this script though, and was only was made a function to keep `main` more streamlined.